easypy/caching.py

```diff
@@ -179,12 +180,14 @@ def inner(*args, **kwargs):
                 result, ts = cache.get(key, NOT_CACHED)
 
                 if expiration > 0:
-                    if result is not NOT_FOUND and time.time() - ts > expiration:
+                    if result is not NOT_FOUND and get_ts_func() - ts > expiration:
                         result = NOT_FOUND
                         del cache[key]
 
                 if result is NOT_FOUND:
-                    ts = time.time()
+                    if log_recalculation:
+                        _logger.debug('time cache expired, calculating new value for %s', name)
+                    ts = get_ts_func()
```

Comment:
Different approach - I would have called it **after** running the function. Since it's a cache we should assume `func()` may take a while, so it makes a difference...